### PR TITLE
Updates to testutils

### PR DESCRIPTION
**Pull request by lornasong**
Various test improvements I came across

Changes:
 - Stop watcher in TestServicesRenderRace
 - Add testutils.Setenv()
 - Tidy up testutils.FreePort()

### CODE REVIEW

File: templates/hcltmpl/integration_test.go [hunk 1]
```diff
@@ -18,20 +17,8 @@ import (
...
-		defer os.Unsetenv("CTS_TEST_ENV")
-	}
-	os.Setenv("CTS_TEST_ENV", "foobar")
-
-	cachedTestDNEEnv, ok := os.LookupEnv("CTS_TEST_DNE")
-	if ok {
-		defer os.Setenv("CTS_TEST_DNE", cachedTestDNEEnv)
-	} else {
-		defer os.Unsetenv("CTS_TEST_DNE")
-	}
```
Comment (lornasong) [inline templates/hcltmpl/integration_test.go:34]: I removed this since it wasn't setting anything but may have misunderstood

Comment (findkim) [reply templates/hcltmpl/integration_test.go:34]: That's fair! `CTS_TEST_DNE` is used in a test case, but it's very unlikely that the host has that environment set to set it back. This was to ensure nothing is currently set before the test case is run, or if it did exist it wouldn't just clean someone's setup from this env.

**Review by findkim — APPROVED** [review]
I love your cleanup PRs. The env cleanup is lovely 😍

File: templates/hcltmpl/integration_test.go (hunk 1 above)
Comment (lornasong) [reply templates/hcltmpl/integration_test.go:34]: Gotcha! Thanks for checking and clarifying

**Pull request merged**